app/main.py

```diff
@@ -1,25 +1,65 @@
 from abc import ABC
+from typing import Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError(f"{self.protected_name} must be an integer.")
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError("Error")
```

Comment:
The error message in the `ValueError` is too generic. Consider providing more specific information about the constraint that was violated, such as the expected range.